src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+  checkIfAlive() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] The checkIfAlive method uses filter inside a loop which can be inefficient if called multiple times (e.g., when multiple animals are bitten in quick succession). Instead of calling checkIfAlive after each bite, consider filtering Animal.alive array once after all bite actions are done, or when necessary, to remove dead animals. This will avoid the potential performance hit of filtering the array multiple times.